### PR TITLE
PKG-5552: Update Safetensors dependency to 0.4.4

### PR DESCRIPTION
**Pull request by jnoller**
safetensors 0.4.4

**Destination channel:** defaults

### Links

- [PKG-5552](https://anaconda.atlassian.net/browse/PKG-5552) 
- [Upstream repository](https://github.com/huggingface/safetensors)
- [Upstream changelog/diff](https://github.com/huggingface/safetensors/releases)
- Relevant dependency PRs:
  - https://github.com/AnacondaRecipes/accelerate-feedstock/pull/5

### Explanation of changes:

- Update safetensors to 0.4.4 - it is a dependency of the llama-cpp-scripts / llama.cpp packaging work.
- Remove the s390x skip as the issue may now be fixed, and lack of an updated build blocks the PR above 

[PKG-5552]: https://anaconda.atlassian.net/browse/PKG-5552?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: acc85dcb09ec5e8aa787f588d7ad4d55c103f31e4ff060e17d92cc0e8b8cac73
+  sha256: 5fe3e9b705250d0172ed4e100a811543108653fb2b66b9e702a088ad03772a07
 build:
   number: 1
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  number: 0
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -21,7 +22,7 @@ requirements:
     - pip
     - python
     - wheel
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
Wheel is not needed with maturin.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: acc85dcb09ec5e8aa787f588d7ad4d55c103f31e4ff060e17d92cc0e8b8cac73
+  sha256: 5fe3e9b705250d0172ed4e100a811543108653fb2b66b9e702a088ad03772a07
 build:
   number: 1
-  skip: True  # [py<38 or s390x]
+  skip: True  # [py<38]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  skip: True  # [py<37]
```
https://github.com/huggingface/safetensors/blob/v0.4.4/bindings/python/pyproject.toml#L3

**Review by ViridianMelody — COMMENTED** [review]
Approved pending the changes suggested by Charles.

**Comment by jnoller** [comment]
Accepted all proposed changes!

**Comment by jnoller** [comment]
@cbouss Looks like removing wheel made the linter unhappy:
```
Linter check found the following problems:
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_c1dcz349w7/clone` failed. (See above for error) The following problems have been found:
===== WARNINGS =====

clone/recipe/meta.yaml:24: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
===== ERRORS =====

clone/recipe/meta.yaml:24: missing_wheel: For pypi packages, wheel should be present in the host section
===== Final Report: =====
1 Error and 1 Warning were found
```

**Comment by skupr-anaconda** [comment]
`{% if not (ppc64le and py>310) %}`
and
`{% endif %}`
can be removed as we don't build on `ppc64le` anymore

**Review by skupr-anaconda — APPROVED** [review]
LGTM but the obsolete `jinja2` stuff

**Comment by cbouss** [comment]
> @cbouss Looks like removing wheel made the linter unhappy:
> 
> ```
> Linter check found the following problems:
> ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_c1dcz349w7/clone` failed. (See above for error) The following problems have been found:
> ===== WARNINGS =====
> 
> clone/recipe/meta.yaml:24: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
> ===== ERRORS =====
> 
> clone/recipe/meta.yaml:24: missing_wheel: For pypi packages, wheel should be present in the host section
> ===== Final Report: =====
> 1 Error and 1 Warning were found
> ```

This is a case where the linter behaviour should be corrected.

**Review by cbouss — APPROVED** [review]

**Comment by jnoller** [comment]
@skupr-anaconda Removed obsolete jinja in https://github.com/AnacondaRecipes/safetensors-feedstock/pull/5/commits/dd5f40c88f1ee7d985b15b0a5fc2dadc79a15266

**Pull request merged**